packages/angular/src/utils/nx-devkit/ast-utils.ts

```diff
@@ -676,6 +676,28 @@ function getListOfRoutes(
   return null;
 }
 
+export function isNgStandaloneApp(tree: Tree, projectName: string) {
+  ensureTypescript();
+  const { tsquery } = require('@phenomnomnominal/tsquery');
+
+  const project = readProjectConfiguration(tree, projectName);
+  const mainFile = project.targets.build.options.main;
+
+  if (project.projectType !== 'application' || !mainFile) {
+    return false;
+  }
+
```

Comment:
We need to handle optional properties in path `project.targets?.build?.options?.main` and it would be better to only require TS and tsquery if needed:

```suggestion
  const project = readProjectConfiguration(tree, projectName);
  const mainFile = project.targets?.build?.options?.main;

  if (project.projectType !== 'application' || !mainFile) {
    return false;
  }

  ensureTypescript();
  const { tsquery } = require('@phenomnomnominal/tsquery');
```